frontend/src/metabase/components/Schedule/constants.ts

```diff
@@ -2,81 +2,145 @@ import { c, t } from "ttag";
-  daily: c("adverb").t`daily`,
-  weekly: c("adverb").t`weekly`,
-  monthly: c("adverb").t`monthly`,
-};
-
 export type Weekday = {
   label: string;
   value: ScheduleDayType;
 };
 
-export const weekdays: Weekday[] = [
-  { label: t`Sunday`, value: "sun" },
-  { label: t`Monday`, value: "mon" },
-  { label: t`Tuesday`, value: "tue" },
-  { label: t`Wednesday`, value: "wed" },
-  { label: t`Thursday`, value: "thu" },
-  { label: t`Friday`, value: "fri" },
-  { label: t`Saturday`, value: "sat" },
-];
+/** These strings are created in a function, rather than in module scope, so that ttag is not called until the locale is set */
+export const getScheduleStrings = () => {
+  const scheduleOptionNames = {
+    // The context is needed because 'hourly' can be an adjective ('hourly rate') or adverb ('update hourly'). Same with 'daily', 'weekly', and 'monthly'.
+    hourly: c("adverb").t`hourly`,
+    daily: c("adverb").t`daily`,
+    weekly: c("adverb").t`weekly`,
+    monthly: c("adverb").t`monthly`,
+  };
+
+  const weekdays: Weekday[] = [
+    {
+      label: c(
+        "Occurs in phrases like 'Send weekly on Sunday'. Only capitalize if necessary",
+      ).t`Sunday`,
+      value: "sun",
+    },
+    {
+      label: c(
+        "Occurs in phrases like 'Send weekly on Monday'. Only capitalize if necessary",
+      ).t`Monday`,
+      value: "mon",
+    },
+    {
+      label: c(
+        "Occurs in phrases like 'Send weekly on Tuesday'. Only capitalize if necessary",
+      ).t`Tuesday`,
+      value: "tue",
+    },
+    {
+      label: c(
+        "Occurs in phrases like 'Send weekly on Wednesday'. Only capitalize if necessary",
+      ).t`Wednesday`,
+      value: "wed",
+    },
+    {
+      label: c(
+        "Occurs in phrases like 'Send weekly on Thursday'. Only capitalize if necessary",
+      ).t`Thursday`,
+      value: "thu",
+    },
+    {
+      label: c(
+        "Occurs in phrases like 'Send weekly on Friday'. Only capitalize if necessary",
+      ).t`Friday`,
+      value: "fri",
+    },
+    {
+      label: c(
+        "Occurs in phrases like 'Send weekly on Saturday'. Only capitalize if necessary",
+      ).t`Saturday`,
+      value: "sat",
+    },
+  ];
 
-export const weekdayOfMonthOptions = [
-  { label: t`calendar day`, value: "calendar-day" },
-  ...weekdays,
-];
+  const weekdayOfMonthOptions: (
+    | Weekday
+    | { label: string; value: "calendar-day" }
+  )[] = [{ label: t`calendar day`, value: "calendar-day" }, ...weekdays];
 
-export const amAndPM = [
-  { label: c("As in 9:00 AM").t`AM`, value: "0" },
-  { label: c("As in 9:00 PM").t`PM`, value: "1" },
-];
+  const amAndPM = [
+    // We use a fallback string in case the translator translated
+    // 'AM' or 'PM' as an empty string, which might happen since
+    // certain cultures do not use AM/PM.
+    { label: c("As in 9:00 AM").t`AM`.trim() || "AM", value: "0" },
+    { label: c("As in 9:00 PM").t`PM`.trim() || "PM", value: "1" },
```

Comment:
I've added this fallback value because the Finnish translation currently uses spaces as the translation for "AM" and "PM", which causes a bug